drivers/include/abp2.h

```diff
@@ -0,0 +1,283 @@
+ * -# Poll the busy flag until it clears with #abp2_getstatus().
+ * -# Retrive the results of the conversion with #abp2_read_raw().
+ *
+ * ### Start measurement and read previous values
+ *
+ * A single call to #abp2_measure_read() will start a conversion of the ADC
+ * and retrieve the results of the \b previous measurement.
+ *
+ * ## Caveats
+ *
+ * This driver currently doesn't support the I2C flavor of these sensors.
+ *
+ * @{
+ *
+ * @file
+ * @brief      Honeywell ABP2 series pressure and temperature
+ *             sensor driver
+ *
+ * @author     David Picard <david.picard@clermont.in2p3.fr>
+ */
+
+
+#ifndef ABP2_H
+#define ABP2_H
+
+#include "saul.h"
+
+#if defined(MODULE_ABP2_SPI)
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#elif defined(MODULE_ABP2_I2C)
+#include "periph/i2c.h"
+#else
+#error "ABP2 driver >> Please add a valid module: either abp2_spi or abp2_i2c"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(MODULE_ABP2_I2C)
+/**
+ * @brief   Default I2C slave address for ABP2 devices
+ *
+ * The sensor address cannot be changed.
+ * A different address can only be obtained upon special request to Honeywell.
+ */
+#define ABP2_ADDR_DEFAULT       (0x28)
+#endif
+
+
+/**
+ * @defgroup abp2_statusbits Status bits
+ * @{
+ *  Macro definitions to manipulate the bits of the status byte.
+ *
+ * Status byte explanation :
+ *
+ *  Bit | Meaning                       | Comment |
+ *  --- | ---                           | --- |
+ *  7   | always 0                      | |
+ *  6   | 1 : device is powered         | |
+ *  5   | 1 : device is busy            | Wait longer between 2 commands. |
+ *  4   | always 0                      | |
+ *  3   | always 0                      | |
+ *  2   | 1 : integrity test failed     | Checksum error at power-up. |
+ *  1   | always 0                      | |
+ *  0   | 1 : internal math saturation  | |
+ */
+#define ABP2_STATUS_BUSY        0x20    /**< Sensor is busy. */
+#define ABP2_STATUS_MEMERR      0x04    /**< Memory integrity error. */
+#define ABP2_STATUS_MATHSAT     0x01    /**< Math saturation error. */
+/** A bitwise AND of the status byte with this mask will return true in case of error. */
+#define ABP2_STATUS_MASK        (ABP2_STATUS_BUSY | ABP2_STATUS_MEMERR | ABP2_STATUS_MATHSAT)
+/** @} */ /* end of abp2_statusbits */
+
+/**
+ * @brief   Parameters for ABP2 sensors.
+ */
+typedef struct abp2_params {
+#if defined(MODULE_ABP2_SPI)
+    spi_t spi;              /**< SPI bus the sensor is connected to. */
+    spi_cs_t cs;            /**< CS pin GPIO handle. */
+    spi_clk_t clk;          /**< SPI bus clock speed. */
+#else
+    i2c_t i2c;              /**< I2C bus the device is connected to. */
+    uint8_t addr;           /**< Device address on the I2C bus. */
+#endif
+    float rangeMin;         /**< Minimum pressure range (sensor unit). */
+    float rangeMax;         /**< Maximum pressure range (sensor unit). */
```

Comment:
Does this have to be `float`?

Reply:
I know floats are not recommended. I don't know how to do otherwise.

Reply:
You can always return a smaller SI unit, e.g. milli °C instead of *C

Reply:
OK. Done.